analyzers/tests/SonarAnalyzer.Test/Trackers/ArgumentTrackerTest.cs

```diff
@@ -0,0 +1,1290 @@
+                        Return 1
+                    End Get
+                    Set(ByVal value As Integer)
+                    End Set
+                End Property
+
+                Public Sub M()
+                    {{access}}
+                End Sub
+            End Class
+            """;
+        var context = ArgumentContextVB(snippet);
+
+        var descriptor = ArgumentDescriptor.ElementAccess(
+            m => m is { MethodKind: var kind, ContainingType: { } type } && type.Name == "C" && (isGetter ? kind == MethodKind.PropertyGet : kind == MethodKind.PropertySet),
+            (n, c) => true,
+            (_, _, _) => true,
+            p => p.Name == parameterName, (_, _) => true);
+        var result = MatchArgumentVB(context, descriptor);
+        result.Should().BeTrue();
+    }
+
+    [DataTestMethod]
+    [DataRow("""process.Modules[$$0]""")]
+    [DataRow("""process?.Modules[$$0]""")]
+    [DataRow("""process.Modules?[$$0]""")]
+    [DataRow("""process?.Modules?[$$0]""")]
+    [DataRow("""process.Modules[index: $$0]""")]
+    [DataRow("""process?.Modules?[index: $$0]""")]
+    public void Indexer_ModulesAccess(string modulesAccess)
+    {
+        var snippet = $$"""
+            public class Test
+            {
+                public void M(System.Diagnostics.Process process)
+                {
+                    _ = {{modulesAccess}};
+                }
+            }
+            """;
+        var context = ArgumentContextCS(snippet);
+
+        var descriptor = ArgumentDescriptor.ElementAccess(m => m is { MethodKind: MethodKind.PropertyGet, ContainingType: { } type } && type.Name == "ProcessModuleCollection",
+            (n, c) => n.Equals("Modules", c), (_, _, _) => true, p => p.Name == "index", (_, p) => p is null or 0);
+        var result = MatchArgumentCS(context, descriptor);
+        result.Should().BeTrue();
+    }
+
+    [DataTestMethod]
+    [DataRow("""processStartInfo.Environment[$$"TEMP"]""")]
+    [DataRow("""processStartInfo?.Environment[$$"TEMP"]""")]
+    [DataRow("""processStartInfo.Environment?[$$"TEMP"]""")]
+    [DataRow("""processStartInfo?.Environment?[$$"TEMP"]""")]
+    [DataRow("""processStartInfo.Environment[key: $$"TEMP"]""")]
+    [DataRow("""processStartInfo?.Environment?[key: $$"TEMP"]""")]
+    public void Indexer_Environment(string environmentAccess)
+    {
+        var snippet = $$"""
+            public class Test
+            {
+                public void M(System.Diagnostics.ProcessStartInfo processStartInfo)
+                {
+                    _ = {{environmentAccess}};
+                }
+            }
+            """;
+        var context = ArgumentContextCS(snippet);
+
+        var descriptor = ArgumentDescriptor.ElementAccess(KnownType.System_Collections_Generic_IDictionary_TKey_TValue, "Environment", p => p.Name == "key", 0);
+        var result = MatchArgumentCS(context, descriptor);
+        result.Should().BeTrue();
+    }
+
+#if NET5_0_OR_GREATER
```

Comment:
can't this be just `#NET`? We run the tests only on net 7 on this branch and on net 8 on master.